<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Mutation type specific overdispersion</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mutation rate modeling</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Mutation type specific overdispersion</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>mutation_rate_modeling/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220224code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220224)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220224code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220224)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongc1eb7b8"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> c1eb7b8 </a>
</p>
</div>
<div id="strongRepositoryversionstrongc1eb7b8" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version c1eb7b8. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  VennDiagram.2022-04-06_15-03-36.log
    Untracked:  VennDiagram.2022-04-13_12-47-23.log
    Untracked:  VennDiagram.2022-04-13_12-51-01.log
    Untracked:  VennDiagram.2022-04-13_12-52-58.log
    Untracked:  VennDiagram.2022-04-13_12-58-02.log
    Untracked:  VennDiagram.2022-04-13_14-58-41.log
    Untracked:  VennDiagram.2022-04-18_01-35-01.log
    Untracked:  VennDiagram.2022-04-18_01-40-42.log
    Untracked:  VennDiagram.2022-04-21_14-57-10.log
    Untracked:  VennDiagram.2022-04-22_12-29-03.log

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
c1eb7b8
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-27
</td>
<td>
wflow_publish(c("analysis", "docs"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
ebc39cc
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
bad560d
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
71b65e8
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-21
</td>
<td>
wflow_publish("analysis/")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
8d95eba
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
e187c35
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-18
</td>
<td>
wflow_publish("analysis/")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
b345803
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
d4a17bc
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-18
</td>
<td>
wflow_publish("analysis/")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
a049b67
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
0a8ec2b
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-13
</td>
<td>
wflow_publish(c("analysis", "docs"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
6077376
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
699c574
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-13
</td>
<td>
wflow_publish(c("analysis", "docs"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
852e6d8
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
e9731b9
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-13
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
1e4c04d
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
82722f8
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-13
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
3978540
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
b8f014a
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-06
</td>
<td>
wflow_publish(c("analysis", "data", "docs"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
a50ea9a
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
10d85d8
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-30
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
fbef0e1
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
8ed415f
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-30
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
7f636da
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
964ae17
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-30
</td>
<td>
wflow_publish(c("analysis", "data", "docs"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
964ae17
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-30
</td>
<td>
wflow_publish(c("analysis", "data", "docs"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
eb6e4e8
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
ad308fe
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-14
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
7feb266
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
4b51b43
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-09
</td>
<td>
wflow_publish(c("analysis", "docs", "data"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
4b51b43
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-09
</td>
<td>
wflow_publish(c("analysis", "docs", "data"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
cd9a5f6
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
a64ace5
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-01
</td>
<td>
wflow_publish(c("analysis", "data"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
f999706
</td>
<td>
Lifan Liang
</td>
<td>
2022-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
f992089
</td>
<td>
Lifan Liang
</td>
<td>
2022-02-27
</td>
<td>
wflow_publish(c("analysis", "data"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
b41be05
</td>
<td>
Lifan Liang
</td>
<td>
2022-02-24
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="problem-overview" class="section level3">
<h3>Problem overview</h3>
<p>For the <span class="math inline">\(i\)</span>th window with a certain size in the genome, we assume the observed mutation rates (<span class="math inline">\(y_i\)</span>) and expected mutation rates (<span class="math inline">\(mu_i\)</span>) were generated with the probabilistic model below: <span class="math display">\[
y_i \sim Poisson(\mu_i\theta_i)
\]</span> <span class="math display">\[
\theta_i \sim Gamma(\alpha,\alpha) 
\]</span></p>
</div>
<div id="mle-of-overdispersion-alpha" class="section level3">
<h3>MLE of overdispersion (<span class="math inline">\(\alpha\)</span>)</h3>
<p>Integrate out <span class="math inline">\(\theta_i\)</span> <span class="math display">\[
P(y_i | \mu_i, \alpha) = \int_0^{+\infty}  P(y_i|\mu_i,\theta_i)P(\theta_i|\alpha) \,d\theta_i
=\dfrac{\mu_i^{y_i} \alpha^\alpha}{y_i!\Gamma(\alpha)}\int_0^{+\infty} \theta_i^{y_i+\alpha-1}e^{-(\mu_i+\alpha)\theta_i}\,d\theta_i
\]</span> The integral part is the same as the kernel of <span class="math inline">\(\theta_i\sim Gamma(y_i+\alpha, \mu_i+\alpha)\)</span>. Thus we got <span class="math display">\[
P(y_i | \mu_i, \alpha)=\dfrac{\mu_i^{y_i} \alpha^\alpha}{y_i!\Gamma(\alpha)}\cdot \dfrac{\Gamma(y_i+\alpha)}{(\mu_i+\alpha)^{y_i+\alpha}}
\]</span> The loglikelihood would be: <span class="math display">\[
\ell(\alpha;y_i,\mu_i) \propto log(\dfrac{\Gamma(y_i+\alpha)}{\Gamma(\alpha)})-y_ilog(\mu_i+\alpha)-\alpha log(1+\mu_i/\alpha)
\]</span> Considering the sparsity and the large sample size (3e10/window_size), we can further simplify the computation by isolating the scenario where <span class="math inline">\(y_i=0\)</span>. When <span class="math inline">\(y_i=0,\mu_i=0\)</span>, <span class="math inline">\(LL\)</span> can be ignored: <span class="math display">\[
\ell(\alpha;y_i,\mu_i) = const
\]</span> When <span class="math inline">\(y_i=0,\mu_i&gt;0\)</span>: <span class="math display">\[
\ell(\alpha;y_i,\mu_i) \propto -\alpha log(1+\mu_i/\alpha)
\]</span></p>
</div>
<div id="validation-of-the-estimator" class="section level3">
<h3>Validation of the estimator</h3>
<p>optim with “BFGS” method was used for MLE of overdispersion. I recomputed <span class="math inline">\(\alpha\)</span> for 5k window mutation rates. Estimated <span class="math inline">\(\alpha\)</span> is 2.709, close to the values in the proposal.</p>
</div>
<div id="asymptotic-normality-of-mle" class="section level3">
<h3>Asymptotic normality of MLE</h3>
<p>The second derivative of <span class="math inline">\(\ell(\alpha;y_i,\mu_i)\)</span> is <span class="math display">\[
\frac{d^2}{d\alpha^2}\ell(\alpha;y_i,\mu_i)=\sum_{k=1}^{y_i}\frac{-1}{(\alpha+k)^2}+\frac{y_i}{(\mu_i+\alpha)^2}+\frac{1}{\alpha}-\frac{\mu_i}{(\alpha+\mu_i)^2}-\frac{1}{\alpha+\mu_i}
\]</span> We used <span class="math inline">\(\frac{d^2}{d\alpha^2}\ell(\alpha;y_i,\mu_i)\)</span> to estimate the standard deviation of <span class="math inline">\(\alpha\)</span>.</p>
</div>
<div id="estimating-alpha-for-different-mutation-types" class="section level3">
<h3>Estimating <span class="math inline">\(\alpha\)</span> for different mutation types</h3>
<ul>
<li>Results here are based on <em>100 bp</em> window size</li>
</ul>
<table>
<thead>
<tr class="header">
<th align="left">mut_type</th>
<th align="right">alpha</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A_to_C</td>
<td align="right">0.0194250</td>
</tr>
<tr class="even">
<td align="left">A_to_G</td>
<td align="right">0.3729952</td>
</tr>
<tr class="odd">
<td align="left">A_to_T</td>
<td align="right">0.0378105</td>
</tr>
<tr class="even">
<td align="left">C_to_A/CpG</td>
<td align="right">3.7335085</td>
</tr>
<tr class="odd">
<td align="left">C_to_A/nonCpG</td>
<td align="right">0.0640994</td>
</tr>
<tr class="even">
<td align="left">C_to_G/CpG</td>
<td align="right">3.7227402</td>
</tr>
<tr class="odd">
<td align="left">C_to_G/nonCpG</td>
<td align="right">0.1168848</td>
</tr>
<tr class="even">
<td align="left">C_to_T/CpG</td>
<td align="right">0.1879970</td>
</tr>
<tr class="odd">
<td align="left">C_to_T/nonCpG</td>
<td align="right">0.2260366</td>
</tr>
</tbody>
</table>
<p>The plot of the gamma distribution. “C_to_A.CpG” and “C_to_G.CpG” are too close to seperate. <img src="figure/index.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
f999706
</td>
<td>
Lifan Liang
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="aggregate-100-bp-to-estimate-alpha-for-5k-bps" class="section level3">
<h3>Aggregate 100 bp to estimate <span class="math inline">\(\alpha\)</span> for 5K bps</h3>
<table>
<thead>
<tr class="header">
<th align="left">mut_type</th>
<th align="right">alpha</th>
<th align="right">std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A_to_C</td>
<td align="right">0.53</td>
<td align="right">24315.78</td>
</tr>
<tr class="even">
<td align="left">A_to_G</td>
<td align="right">23.33</td>
<td align="right">4.43</td>
</tr>
<tr class="odd">
<td align="left">A_to_T</td>
<td align="right">1.28</td>
<td align="right">3041.25</td>
</tr>
<tr class="even">
<td align="left">C_to_A_CpG</td>
<td align="right">17.12</td>
<td align="right">0.38</td>
</tr>
<tr class="odd">
<td align="left">C_to_A_nonCpG</td>
<td align="right">27.74</td>
<td align="right">0.89</td>
</tr>
<tr class="even">
<td align="left">C_to_G_CpG</td>
<td align="right">1.40</td>
<td align="right">303.61</td>
</tr>
<tr class="odd">
<td align="left">C_to_G_nonCpG</td>
<td align="right">1.02</td>
<td align="right">6838.35</td>
</tr>
<tr class="even">
<td align="left">C_to_T_CpG</td>
<td align="right">1.67</td>
<td align="right">4263.19</td>
</tr>
<tr class="odd">
<td align="left">C_to_T_nonCpG</td>
<td align="right">44.40</td>
<td align="right">0.61</td>
</tr>
</tbody>
</table>
<p><img src="figure/index.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4b51b43
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-09
</td>
</tr>
<tr>
<td>
cd9a5f6
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="observed-rates-vs.-expected-rates" class="section level3">
<h3>Observed rates vs. expected rates</h3>
<div id="correlation-at-5kb-level" class="section level4">
<h4>Correlation at 5kb level</h4>
<p>The correlation between expected rates and observed rates is probably weak. This is consistent across mutation types (the figure on the top) and the total rate counts (the bottom figure). <img src="assets/exp_vs_obs_mut_type.png" /> <img src="assets/exp_vs_obs_total.png" /></p>
</div>
<div id="correlation-at-chromosome-level" class="section level4">
<h4>Correlation at chromosome level</h4>
<p>There is strong correlation when the mutation counts are summed up to the chromosome level. <img src="assets/exp_vs_obs_chr_level.png" /></p>
</div>
<div id="total-counts-for-each-mutation-type" class="section level4">
<h4>Total counts for each mutation type</h4>
<p>The total counts are also quite close between observed and expected.</p>
<table>
<thead>
<tr class="header">
<th align="left">mutation type</th>
<th align="left">observed counts</th>
<th align="left">expected counts</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A_to_C_and_5k.stat</td>
<td align="left">7799.0</td>
<td align="left">9107.146427551219</td>
</tr>
<tr class="even">
<td align="left">C_to_T_nonCpG_5k.stat</td>
<td align="left">27716.0</td>
<td align="left">30422.830678788276</td>
</tr>
<tr class="odd">
<td align="left">C_to_A_CpG_5k.stat</td>
<td align="left">1052.0</td>
<td align="left">932.9034470382618</td>
</tr>
<tr class="even">
<td align="left">C_to_G_nonCpG_5k.stat</td>
<td align="left">9799.0</td>
<td align="left">10088.667585756764</td>
</tr>
<tr class="odd">
<td align="left">A_to_T_and_5k.stat</td>
<td align="left">7407.0</td>
<td align="left">7767.137108131126</td>
</tr>
<tr class="even">
<td align="left">C_to_A_nonCpG_5k.stat</td>
<td align="left">10002.0</td>
<td align="left">10402.731650556543</td>
</tr>
<tr class="odd">
<td align="left">A_to_G_and_5k.stat</td>
<td align="left">30001.0</td>
<td align="left">30685.1840776996</td>
</tr>
<tr class="even">
<td align="left">C_to_G_CpG_5k.stat</td>
<td align="left">899.0</td>
<td align="left">781.5190549596658</td>
</tr>
<tr class="odd">
<td align="left">C_to_T_CpG_5k.stat</td>
<td align="left">20166.0</td>
<td align="left">20643.965893071087</td>
</tr>
</tbody>
</table>
<p><img src="assets/mut_type_total.png" /></p>
</div>
</div>
<div id="definition-of-hotspots" class="section level3">
<h3>Definition of hotspots</h3>
<p><span class="math inline">\(\ell(\alpha;y_i,\mu_i)\)</span> is actually the normalization factor for <span class="math inline">\(P(\theta|y_i,\mu_i,\alpha)\)</span>, the posterior <span class="math inline">\(theta\)</span> is <span class="math inline">\(Gamma(y_i+\alpha, \mu_i+\alpha)\)</span>. Hence we have <span class="math display">\[
E(\theta|y_i,\mu_i,\alpha)=\frac{y_i+\alpha}{\mu_i+\alpha}
\]</span></p>
<p>Previously, I tried to define hotspots based on <span class="math inline">\(E(\theta|y_i,\mu_i,\alpha)\)</span>. As shown in the figure below, large <span class="math inline">\(\alpha\)</span> in some mutation types may make it difficult to find hotspots. <img src="assets/theta_post_mut_type.png" /></p>
<p>Alternatively, we detect hotspots with a simple test. Say the null model is that: <span class="math display">\[
y_i \sim Poisson(\mu_i)
\]</span> And the alternative is that <span class="math inline">\(y_i\)</span> is not sampled from the Poisson of <span class="math inline">\(\mu_i\)</span>. We used <span class="math inline">\(P&lt;0.05\)</span> as the criteria.</p>
</div>
<div id="hotspots-continuity" class="section level3">
<h3>Hotspots continuity</h3>
<p>I computed how many hotspot windows are actually adjacent to another (“continuous” column in the count table). C_to_T and A_to_G have the most hotspots.</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">total</th>
<th align="right">hotspots</th>
<th align="right">continuous</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A_to_C_and_5k</td>
<td align="right">576194</td>
<td align="right">76</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">C_to_T_nonCpG_5k</td>
<td align="right">576194</td>
<td align="right">164</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">C_to_A_CpG_5k</td>
<td align="right">576194</td>
<td align="right">5</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">C_to_G_nonCpG_5k</td>
<td align="right">576194</td>
<td align="right">170</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">A_to_T_and_5k</td>
<td align="right">576194</td>
<td align="right">81</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">C_to_A_nonCpG_5k</td>
<td align="right">576194</td>
<td align="right">124</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A_to_G_and_5k</td>
<td align="right">576194</td>
<td align="right">62</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">C_to_G_CpG_5k</td>
<td align="right">576194</td>
<td align="right">177</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">C_to_T_CpG_5k</td>
<td align="right">576194</td>
<td align="right">254</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>Then, we looked into the distance from each hotspot to its closest hotspot window. Hotspots in C_to_T and A_to_G have smaller distances. This result is consistent with the count table above <img src="assets/hotspot_dist.png" />.</p>
</div>
<div id="hotspot-overlap-across-mutation-types-one-5k-window-away" class="section level3">
<h3>Hotspot overlap across mutation types (one 5k window away)</h3>
<p>Here, we define the overlap as two hotspot windows are one window away. Jaccard coefficient was computed. <span class="math display">\[
Jaccard(A,B) = \dfrac{|A\cap B|}{|A\cup B|}
\]</span></p>
<p><img src="figure/index.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
b345803
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-18
</td>
</tr>
<tr>
<td>
3978540
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-06
</td>
</tr>
<tr>
<td>
964ae17
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>As shown in the heatmap, A_to_G, C_to_T_CpG, and C_to_T_nonCpG share more hotspots. Hotspots in C_to_G_CpG and C_to_A_CpG are not similar to other mutation types.</p>
<p><img src="figure/index.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
b345803
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-18
</td>
</tr>
<tr>
<td>
3978540
</td>
<td>
Lifan Liang
</td>
<td>
2022-04-06
</td>
</tr>
<tr>
<td>
964ae17
</td>
<td>
Lifan Liang
</td>
<td>
2022-03-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="bayesian-poisson-regression" class="section level2">
<h2>Bayesian Poisson regression</h2>
<p>Given observed mutation rates (<span class="math inline">\(y_i\)</span>), baseline mutation rates (<span class="math inline">\(\mu_i\)</span>) and 14 genomic features (<span class="math inline">\(X_i\)</span>). The generative process is</p>
<p><span class="math display">\[
\tau^2_j \sim Inv-\chi^2(1,1)
\]</span> <span class="math display">\[
\beta_{jt} \sim N(0,\tau^2_j)
\]</span> <span class="math display">\[
\theta_{it} = exp(\beta_{0t}+\sum_j \beta_{jt}X_{ij})
\]</span></p>
<p><span class="math display">\[
Y_{it} \sim Pois(\mu_i \cdot \theta_{it})
\]</span></p>
<p>Numpyro MCMC was used to infer the posterior of <span class="math inline">\(\beta_{jt}\)</span>.</p>
<p>I initialized four chains. For each chain, 2000 samples were drawn after warmup of 1000 samples. Chains seem mixed.</p>
<p><img src="assets/numpyro_mixing.png" /></p>
<p><img src="assets/beta_jt_numpyro.png" />.</p>
<div id="stan-implementation" class="section level3">
<h3>Stan implementation</h3>
<p>The same model was implemented with rstan. The NUTS sampler was used. I initiated 4 chains</p>
<div id="diagnostics" class="section level4">
<h4>Diagnostics</h4>
<p>First, the four chains were well mixed, as shown below.</p>
<p><img src="assets/stan_trace.png" /></p>
<p>Next, the energy distribution was close to the marginal energy distribution. According to Stan user guide, this is a good sign that the sampler explored the target distribution efficiently.</p>
<p><img src="assets/stan_diag_energy.png" /></p>
<p>We further performed the posterior predictive check. Essentially, we draw a vector of <span class="math inline">\(\beta\)</span> from the posterior and draw <span class="math inline">\(\tilde{y}\)</span> from the Poisson distribution parameterized by <span class="math inline">\(\beta\)</span>. Then we compared the distribution of <span class="math inline">\(\tilde{y}\)</span> and observed <span class="math inline">\(y\)</span>.</p>
<p>As shown in the figure below, <span class="math inline">\(\tilde{y}\)</span> is close to <span class="math inline">\(y\)</span> across mutation types. <img src="assets/post_pred_checks.png" /></p>
</div>
<div id="parameter-estimation" class="section level4">
<h4>Parameter estimation</h4>
<p><img src="assets/stan_beta_est.png" /></p>
</div>
<div id="cross-validation-with-log-pointwise-posterior-probability-lpd" class="section level4">
<h4>Cross validation with log pointwise posterior probability (lpd)</h4>
<p>The idea is to estimate parameters with one set of windows and estimate the fitness in another set of windows. According to Stan user guide, lpd can be computed as:</p>
<p><span class="math display">\[
lpd=\frac{1}{N}\sum_{n=1}^N{log\,p(y_n|y,M_1)}
\]</span> where <span class="math inline">\(p(y_n|y,M_1)\)</span> can be obtained by averaging over posterior predictive. <span class="math display">\[
p(y_n|y,M_1)=\int{P(y_n|\theta)P(\theta|y,M_1)}\,d\theta=E_{\theta|y}(l(\theta;y_n))
\]</span></p>
<p>Suppose we already obtained posterior samples of <span class="math inline">\(\tau\)</span> and <span class="math inline">\(\beta\)</span> from the odd windows. To estimate lpd in the even windows,</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ChIPpeakAnno_3.28.1  rtracklayer_1.54.0   GenomicRanges_1.46.1
 [4] GenomeInfoDb_1.30.1  IRanges_2.28.0       S4Vectors_0.32.3    
 [7] BiocGenerics_0.40.0  RColorBrewer_1.1-2   ggplot2_3.3.5       
[10] workflowr_1.7.0     

loaded via a namespace (and not attached):
  [1] colorspace_2.0-2            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.2            
  [5] futile.logger_1.4.3         XVector_0.34.0             
  [7] fs_1.5.2                    rstudioapi_0.13            
  [9] farver_2.1.0                bit64_4.0.5                
 [11] AnnotationDbi_1.56.2        fansi_1.0.2                
 [13] xml2_1.3.3                  splines_4.1.2              
 [15] cachem_1.0.6                knitr_1.37                 
 [17] Rsamtools_2.10.0            dbplyr_2.1.1               
 [19] png_0.1-7                   graph_1.72.0               
 [21] compiler_4.1.2              httr_1.4.2                 
 [23] assertthat_0.2.1            Matrix_1.3-4               
 [25] fastmap_1.1.0               lazyeval_0.2.2             
 [27] cli_3.1.1                   later_1.3.0                
 [29] formatR_1.11                htmltools_0.5.2            
 [31] prettyunits_1.1.1           tools_4.1.2                
 [33] gtable_0.3.0                glue_1.6.1                 
 [35] GenomeInfoDbData_1.2.7      dplyr_1.0.7                
 [37] rappdirs_0.3.3              Rcpp_1.0.8                 
 [39] Biobase_2.54.0              jquerylib_0.1.4            
 [41] vctrs_0.3.8                 Biostrings_2.62.0          
 [43] multtest_2.50.0             xfun_0.29                  
 [45] stringr_1.4.0               ps_1.6.0                   
 [47] lifecycle_1.0.1             restfulr_0.0.13            
 [49] ensembldb_2.18.3            XML_3.99-0.8               
 [51] InteractionSet_1.22.0       MASS_7.3-54                
 [53] getPass_0.2-2               zlibbioc_1.40.0            
 [55] scales_1.1.1                BSgenome_1.62.0            
 [57] vroom_1.5.7                 hms_1.1.1                  
 [59] promises_1.2.0.1            MatrixGenerics_1.6.0       
 [61] ProtGenerics_1.26.0         parallel_4.1.2             
 [63] SummarizedExperiment_1.24.0 RBGL_1.70.0                
 [65] AnnotationFilter_1.18.0     lambda.r_1.2.4             
 [67] yaml_2.2.2                  curl_4.3.2                 
 [69] memoise_2.0.1               biomaRt_2.50.3             
 [71] stringi_1.7.6               RSQLite_2.2.9              
 [73] highr_0.9                   BiocIO_1.4.0               
 [75] GenomicFeatures_1.46.4      filelock_1.0.2             
 [77] BiocParallel_1.28.3         rlang_1.0.0                
 [79] pkgconfig_2.0.3             matrixStats_0.61.0         
 [81] bitops_1.0-7                evaluate_0.14              
 [83] lattice_0.20-45             purrr_0.3.4                
 [85] GenomicAlignments_1.30.0    labeling_0.4.2             
 [87] bit_4.0.4                   processx_3.5.2             
 [89] tidyselect_1.1.1            magrittr_2.0.2             
 [91] R6_2.5.1                    generics_0.1.2             
 [93] DelayedArray_0.20.0         DBI_1.1.2                  
 [95] pillar_1.7.0                whisker_0.4                
 [97] withr_2.4.3                 survival_3.2-13            
 [99] KEGGREST_1.34.0             RCurl_1.98-1.6             
[101] tibble_3.1.6                crayon_1.4.2               
[103] futile.options_1.0.1        utf8_1.2.2                 
[105] BiocFileCache_2.2.1         tzdb_0.2.0                 
[107] rmarkdown_2.11              progress_1.2.2             
[109] grid_4.1.2                  blob_1.2.2                 
[111] callr_3.7.0                 git2r_0.29.0               
[113] digest_0.6.29               VennDiagram_1.7.1          
[115] regioneR_1.26.1             httpuv_1.6.5               
[117] munsell_0.5.0              </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
